tools/bazel/compile_commands/compile_commands.py

```diff
@@ -0,0 +1,255 @@
+    bazel = arguments[0]
+    logging.debug("Bazel: %s", bazel)
+    logging.debug("Bazel build arguments:")
+    for argument in arguments:
+        logging.debug("   %s", argument)
+        if re.search(r"^(@|:|\/\/)\S*", argument) or argument == "...":
+            targets.append(argument)
+        elif re.search(r"^--\S*", argument):
+            configs.append(argument)
+    logging.debug("Targets: %s", targets)
+    logging.debug("Configs: %s", configs)
+    if len(targets) < 1:
+        logging.error("No targets found for bazel query in: %s", build_arguments)
+        return
+
+    deps = " + ".join(["deps(\"{target}\")".format(target=target) for target in targets])
+    logging.debug("Deps: %s", deps)
+
+    options = [
+        "--output=jsonproto",
+        "--include_artifacts=false",
+    ]
+    command = "{bazel} aquery {options} 'mnemonic(\"CppCompile\", {deps})' {configs}".format(
+        bazel=bazel, deps=deps, options=" ".join(options), configs=" ".join(configs))
+    logging.debug("Command: %s", command)
+
+    out = run_command(command)
+    if out:
+        data = json.loads(out)
+        return data
+    else:
+        return False
+
+
+def get_compile_commands(data, directories=["."], base_dir=None, output_base=None):
+    """
+    Produce compile commands from action graph JSON data
+    """
+    compile_commands = []
+    not_found = 0
+    skipped = 0
+    for action in data["actions"]:
+        if "mnemonic" in action and action["mnemonic"] in ["CppCompile"]:
+            arguments = action["arguments"]
+            try:
+                index = arguments.index("-c")
+                file = arguments[index + 1]
+                if not file.endswith(".asm"):
+                    (full_path, directory) = get_full_path(file, directories)
+                    if full_path and directory:
+                        if base_dir:
+                            directory = base_dir
+                        compile_command = {
+                            "file": full_path,
+                            "directory": directory,
+                            "command": " ".join(arguments),
+                        }
+                        filter_compile_command(compile_command, output_base)
+                        compile_commands.append(compile_command)
+                        logging.info("Add file: %s", file)
+                    else:
+                        logging.warning("Cannot find: %s", file)
+                        not_found += 1
+                else:
+                    logging.info("Skipping: %s", file)
+                    skipped += 1
+            except ValueError:
+                logging.warning("Wrong compile command: %s", str(arguments))
+    logging.info("%d files added, %d skipped (asm), %d not found",
+                 len(compile_commands), skipped, not_found)
+    return compile_commands
+
+
+def get_full_path(file, directories):
+    """
+    Return full path for a file and try to prepend directories
```

Comment:
This function will return not only with the full path but the parent diectory too. It would be better to return only with the full path or None and use the `os.path.dirname` to get the parent directory for the file if its not None on the call side.

Reply:
Done:

- Changed the name and docstring of the function since `get_full_path` is meisleading